forem/forem
# Auto-generate CSS utility classes documentation in Storybook

## What type of PR is this? (check all applicable)


## Description

CSS utility classes are generated by the https://github.com/forem/forem/blob/master/app/assets/stylesheets/config/_generator.scss file. This PR generates CSS from that file and parses the CSS style sheet to auto-generate Storybook stories to provide documentation for CSS utility classes.

CSS utility classes are grouped by the property they modify. In some cases, some utility classes modify more than one CSS property. In cases like that, the utility class is grouped in each CSS property it modifies.

CSS utility classes will typically have values set with the `!important` attribute so as to work with pre-design system CSS.

Also noteworthy is CSS properties in the auto-generated documentation point to MDN.

![image](https://user-images.githubusercontent.com/833231/89460185-e0a05b00-d737-11ea-99aa-fb7ed331d4f0.png)

## Related Tickets & Documents

## QA Instructions, Screenshots, Recordings

![image](https://user-images.githubusercontent.com/833231/89459514-be5a0d80-d736-11ea-90cd-61da38e490c6.png)

To test:
- Navigate to the [Storybook](https://deploy-preview-9633--storybookdevto.netlify.app/?path=/story/5-css-utility-classes-align-content--content-start) for this PR.
- Section 5 of the Storybook should contain the auto-generated Storybook stories for all the CSS properties that the CSS utility classes modify.
- CSS utilty classes that modify more than one CSS property should appear under each of those CSS properties (the way the Storybook stories are classified for CSS utility classes). For example, the `px-0` utility class appears under `padding-left` and `padding-right`.
  - https://deploy-preview-9633--storybookdevto.netlify.app/?path=/story/5-css-utility-classes-padding-left--px-0
  - https://deploy-preview-9633--storybookdevto.netlify.app/?path=/story/5-css-utility-classes-padding-right--px-0
- The auto-generatated documentation should contain links that redirect to MDN for the CSS properties that are modified by the CSS utility class.

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Mr. Krabs from Sponge Bob saying "I'll show you automated!"](https://media.giphy.com/media/l1KtYG8BndKBmWrM4/giphy.gif)
